src/server.cpp

```diff
@@ -2722,6 +2767,82 @@ void Server::UpdateCrafting(Player* player)
 	plist->changeItem(0, preview);
 }
 
+bool Server::handleChat(const std::string &name, const std::wstring &wname,
+	const std::wstring &wmessage, std::wstring &answer_to_sender,
+	u16 peer_id_to_avoid_sending)
+{
+	// If something goes wrong, this player is to blame
+	RollbackScopeActor rollback_scope(m_rollback,
+		std::string("player:") + name);
+
+	// Line to send
+	std::wstring line;
+	// Whether to send to the player that sent the line, or to all players
+	bool send_to_sender_only = false;
+
+	// Run script hook
+	bool ate = m_script->on_chat_message(name,
+		wide_to_utf8(wmessage));
+	// If script ate the message, don't proceed
+	if (ate)
```

Comment:
Couldn't you eschew this "ate" variable altogether by checking the result of on_chat_message directly?